### ISSUE TITLE
app_router.gr.dart - Too many positional arguments: 0 allowed, but 1 found

### ISSUE THREAD

**Issue by dthuesen** [Milad-Akarie/auto_route_library#1482]
I'm trying to use AutoRoute in the new version 6 but get stuck with an error: "lib/presentation/routes/app_router.gr.dart:62:14: Error: Too many positional arguments: 0 allowed, but 1 found."

This error appears with every Version from 6.0.1 to 6.1.0.

The header of the generated router looks like this (have a look at the third line which reads **_':super(navigatorKey)'_**:

```dart
abstract class $AppRouter extends _i22.RootStackRouter {
  $AppRouter([_i23.GlobalKey<_i23.NavigatorState>? navigatorKey])
      : super(navigatorKey);
...
...
...
```
To get rid of this error I have to change the positional argument to named argument, like this:
```dart
: super(navigatorKey: navigatorKey);
```

I did all the steps to have a clean setup:
- run flutter doctor -> no issues
- deleted app_router.gr.dart and rerun build_runner
- run flutter clean and then pub get commands
- not only did hot restart, did stop app and then run app

None of these steps showed a different result. I think it's an issue in the library.

I also have another issue with this version. I updated from 4.x to 6.x an made all the necessary steps for the migration. With the old version the routing worked like a charm but now I have big trouble with the nested routes I use. The structure of the navigation is like this:
```dart
AutoRoute(
    path: '/', 
    page: WrapperPageRoute.page, 
    guards: [AuthGuard()], 
    children: [
        RedirectRoute(path: '', redirectTo: RouteNames.planning),
        AutoRoute(
            path: RouteNames.planning,
            page: VisualPlanningRoute.page,
            children: [
                AutoRoute(
                    path: RouteNames.visualPlanningTreatmentsDetails,
                    page: TreatmentDetailsRoute.page,
                ),
            ],
        ),   
    ]),
    AutoRoute(
          path: RouteNames.treatments,
          page: TreatmentsOverviewRoute.page,
          children: [
            AutoRoute(
              path: RouteNames.treatmentSearch,
              page: TreatmentsSearchRoute.page,
            ),
            AutoRoute(
              path: RouteNames.treatmentDetails,
              page: TreatmentDetailsRoute.page,
            ),
            AutoRoute(
              path: RouteNames.treatmentVisualPlanning,
              page: VisualPlanningRoute.page,
            ),
          ],
        ),
...
...
...
```
I'm not able to navigate (by button) from `VisualPlanningRoute` to `TreatmentDetailsRoute` or from `TreatmentsOverviewRoute` to `TreatmentsSearchRoute`.

I've tried all available navigation patterns (I think), what would would be your advise to achieve this. The documentation gives not enough support for that.

**Comment by Milad-Akarie** [comment]
Hello @dthuesen  regarding the navigator key issue you need upgrade both auto_route and auto_route_generator to v6.1.0
`VisualPlanningScreen` must have an AutoRouter widget inside of it to render nested routes. is that the case?

**Comment by awaik** [comment]
I had the same issue.
Fixed by changing `auto_route_generator: any` to `auto_route_generator: ^6.1.0`.

**Comment by guplem** [comment]
I have just updated from `^6.0.3` to v. `^6.3.0` and found this issue.
Solved by updating auto_route_generator to `^6.2.0`
